strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProvider.java

```diff
@@ -250,14 +251,7 @@ protected OutputStream getOutputStreamInternal(RepositoryPath repositoryPath)
 
     private ArtifactCoordinates getArtifactCoordinates(Path p)
     {
-        try
-        {
-            return RepositoryFiles.readCoordinates((RepositoryPath) p);
-        }
-        catch (IOException e)
-        {
-            throw new RuntimeException(String.format("Failed to resolve ArtifactCoordinates for [%s]", p), e);
-        }
+        return ThrowingFunction.unchecked(RepositoryFiles::readCoordinates).apply((RepositoryPath) p);
```

Comment:
@adavid9 can you please add `throws IOException` here and use `TrowingFunction` at `GroupRepositoryProvider.search` ?

Reply:
@sbespalov To ensure that I understand it correctly, I should remove `ThrowingFunction` from this place and left it as it was before, but use `ThrowingFunction` in `GroupRepositoryProvider.search`?

Reply:
yes, but you should also throw `IOException` instead of `RuntimeException`